gtsam/CMakeLists.txt

```diff
@@ -175,6 +175,8 @@ target_include_directories(gtsam SYSTEM BEFORE PUBLIC
   # CCOLAMD
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/CCOLAMD/Include>
   $<INSTALL_INTERFACE:include/gtsam/3rdparty/CCOLAMD>
+  # Eigen
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/Eigen>
```

Comment:
Explain?

Reply:
The TLDR is that by marking Eigen as a system include, our compiler will ignore warnings generated by files inside this directory. Without this we do indeed see warnings generated by Eigen, which would be elevated to errors by Werror.

Reply:
From the [horses mouth](https://cmake.org/cmake/help/latest/command/include_directories.html)

> If the SYSTEM option is given, the compiler will be told the directories are meant as system include directories on some platforms. Signaling this setting might achieve effects such as the compiler skipping warnings, or these fixed-install system files not being considered in dependency calculations - see compiler docs

Reply:
Oh great; I'll remove it entirely then